core/src/main/scala/cats/Traverse.scala

```diff
@@ -33,6 +35,17 @@ import simulacrum.typeclass
   def traverse[G[_]: Applicative, A, B](fa: F[A])(f: A => G[B]): G[F[B]]
 
   /**
+   * Traverse with a function returning state.
+   *
+   * Given a function which returns state, thread the state through
+   * the running of this function on all the values in F, returning
+   * F[B] in the state context.
+   */
+  def stateTraverse[S, A, B](fa: F[A])(f: A => State[S, B]): State[S, F[B]] =
+    State[S, F[B]](s =>
+      traverse[State[S, ?] , A, B](fa)(f).run(s).value)
```

Comment:
We don't need the additional `State` wrapping of `run(s).value`, right?

This is equal to `traverse[State[S, ?] , A, B](fa)(f)` and with SI 2712 fixed, we don't even need to specify the types, so it is just `traverse(fa)(f)`.